sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,185 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for pytorch-forecasting models."""
+import abc
+import functools
+import typing
+from typing import Any, Dict, List
+
+import lightning.pytorch as pl
+import pandas
+from pytorch_forecasting.data import TimeSeriesDataSet
+from pytorch_forecasting.metrics import MultiHorizonMetric
+from torch import nn
```

Comment:
Imports of soft dependencies (all of these except `pandas`) should not be present at top level. These gets executed during package imports, irrespective of which components are being used by users. These must be imported inside functions after ensuring they are installed.

Not revieweing rest of the PR, but this is the main reason of CI failures.

Reply:
Thanks for the review. I have moved those imports into functions.